app/components/Sidebar.tsx

```diff
@@ -88,20 +88,30 @@ export const NavLinkItem = (props: {
   children: React.ReactNode
   end?: boolean
   disabled?: boolean
-}) => (
-  <li>
-    <NavLink
-      to={props.to}
-      className={({ isActive }) =>
-        cn(linkStyles, {
-          'text-accent !bg-accent-secondary hover:!bg-accent-secondary-hover svg:!text-accent-tertiary':
-            isActive,
-          'pointer-events-none text-disabled': props.disabled,
-        })
-      }
-      end={props.end}
-    >
-      {props.children}
-    </NavLink>
-  </li>
-)
+}) => {
+  // "New" resource create forms have a url that doesn't match the top-level resource name, so `isActive` won't ever fire as true.
+  // Determine which resource in the Sidebar should be highlighted as active when on a create-form page.
+  const location = useLocation()
+  const resourcePath = location.pathname.split('/').pop()
+  const resourceName = resourcePath?.includes('-new')
+    ? resourcePath.replace('-new', '')
+    : ''
+  const isLinkToThisResource = resourceName.length > 0 && props.to.includes(resourceName)
```

Comment:
You should extract this chunk of logic into a util function that takes `to` and `pathname` and returns the bool, then you can unit test it